install/kubernetes/helm/istio-remote/templates/sidecar-injector-configmap.yaml

```diff
@@ -54,10 +54,15 @@ data:
             - NET_ADMIN
           privileged: true
         restartPolicy: Always
-      {{ if eq .Values.global.proxy.enableCoreDump true }}
+      {{- if and (.Values.global.proxy.enableCoreDump) (ne .Values.global.proxy.maxNumCoreDumps "0") }}
+
       - args:
         - -c
-        - sysctl -w kernel.core_pattern=/etc/istio/proxy/core.%e.%p.%t && ulimit -c unlimited
+      {{ if eq .Values.global.proxy.maxNumCoreDumps "unlimited" }}
```

Comment:
I don't get it - if it is set to 5, how will the pattern work to generate 5 distinct files ? 

IMHO again we are adding too many options and complexity. Just get rid of 'unlimited' (it will not work anyways - the disk is not 'unlimited'), get rid of maxNumCoreDumps - and just generate one core dump until we have clear evidence we need more.

Reply:
The documentation will state (it does already in values.yaml) that only 0,1 and unlimited are currently supported and explains what will happen with different N (it will go to 1).
The reason to have the option is not to take away functionality i.e. some users may want to have unlimited cores that is the old behavior, and also pave the way for the code based core management we've been discussing that prunes old cores, limits space etc.
As usual, there are objections to taking either approach. But if we believe that we want core management in the future I think this is a reasonable API change to get there.